specs/metered-guidance-errors.md

```diff
@@ -0,0 +1,111 @@
+# Metered APIs guidance and random errors
+
+Using Metered plugin, developers can ensure are aware and utilize the Metered APIs in the most efficient way possible.
+
+Metered APIs allows you to get access to advanced capabilities in Microsoft Graph. They require an active Azure subscription with the calling application. Some of these metered APIs are protected and require additional validation beyond permissions and admin consent. The main goal of this guidance is to know that the APIs they are calling would require extra action when deployed in production and could incur a cost.
+
+## History
+
+| Version | Date | Comments | Author |
+| ------- | ---- | -------- | ------ |
+| 1.0 | 2023-03-20 | Initial specifications | @sebastienlevert |
+
+## Configuration
+
+N/A
+
+## Implementation
+
+> **Note**
+> This metered APIs plugin is only valid for Microsoft Graph requests.  
+
+### Guidance
+Metered APIs guidance will be implemented as a plugin, subscribed to one events: `BeforeRequest`.
+
+In the `BeforeRequest` event, the plugin will check if the request URL is part of the identified metered APIs. This list of APIs is documented in the official Microsoft Graph Docs [here](https://learn.microsoft.com/en-us/graph/metered-api-list?view=graph-rest-1.0). This list will likely grow but as a first step, we will be using a static configuration file to identify the endpoints to mark as metered. If the request is targeted at a metered API, we will emit a log message that will include text telling the developer the endpoint is metered and will include a link to the Metered APIs and services in Microsoft Graph documentation: `https://learn.microsoft.com/en-us/graph/metered-api-list`.
+
+```shell
+[ REQUEST ]    GET hhttps://graph.microsoft.com/v1.0/teams/*/channels/getAllMessages
+[   API   ]  ┌ Passed through
+             └ GET https://graph.microsoft.com/v1.0/teams/*/channels/getAllMessages
+[ WARNING ]  ┌ This is a metered API.
+             │ More info at https://learn.microsoft.com/en-us/graph/teams-licenses

```

Comment:
Let's mention explicitly that this URL is coming from the `documentationUrl` config property